digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala

```diff
@@ -0,0 +1,682 @@
+
+  def createValidPeriod(trafficSignType: TrafficSignType, additionalPanel: AdditionalPanel): Set[ValidityPeriod] = {
+    TimePeriodClass.fromTrafficSign(trafficSignType).filterNot(_ == TimePeriodClass.Unknown).flatMap { period =>
+      val regexMatch = "[(]?\\d+\\s*[-]{1}\\s*\\d+[)]?".r
+      val validPeriodsCount = regexMatch.findAllIn(additionalPanel.panelInfo)
+      val validPeriods = regexMatch.findAllMatchIn(additionalPanel.panelInfo)
+
+      if (validPeriodsCount.length == 3 && ValidityPeriodDayOfWeek.fromTimeDomainValue(period.value) == ValidityPeriodDayOfWeek.Sunday) {
+        val convertPeriod = Map(0 -> ValidityPeriodDayOfWeek.Weekday, 1 -> ValidityPeriodDayOfWeek.Saturday, 2 -> ValidityPeriodDayOfWeek.Sunday)
+        validPeriods.zipWithIndex.map { case (timePeriod, index) =>
+          val splitTime = timePeriod.toString.replaceAll("[\\(\\)]|\\s", "").split("-")
+          ValidityPeriod(splitTime.head.toInt, splitTime.last.toInt, convertPeriod(index))
+        }.toSet
+
+      } else
+        validPeriods.map { timePeriod =>
+          val splitTime = timePeriod.toString.replaceAll("[\\(\\)]|\\s", "").split("-")
+          ValidityPeriod(splitTime.head.toInt, splitTime.last.toInt, ValidityPeriodDayOfWeek.fromTimeDomainValue(period.value))
+        }
+    }
+  }
+
+  def segmentsManager(roadLinks: Seq[RoadLink], trafficSigns: Seq[PersistedTrafficSign], existingSegments: Seq[TrafficSignToLinear]): Set[TrafficSignToLinear] = {
+    val startEndRoadLinks = findStartEndRoadLinkOnChain(roadLinks)
+
+    val newSegments = startEndRoadLinks.flatMap { case (roadLink, startPointOfInterest, lastPointOfInterest) =>
+      baseProcess(trafficSigns, roadLinks, roadLink, (startPointOfInterest, lastPointOfInterest, None), Seq())
+    }.distinct
+
+    val groupedAssets = (newSegments ++ existingSegments).groupBy(_.roadLink)
+    val assets = fillTopology(roadLinks, groupedAssets)
+
+    convertEndRoadSegments(assets, startEndRoadLinks).toSet
+  }
+
+  def fillTopology(topology: Seq[RoadLink], linearAssets: Map[RoadLink, Seq[TrafficSignToLinear]]): Seq[TrafficSignToLinear] = {
+    val fillOperations: Seq[Seq[TrafficSignToLinear] => Seq[TrafficSignToLinear]] = Seq(
+      combine,
+      convertOneSideCode
+    )
+
+    topology.foldLeft(Seq.empty[TrafficSignToLinear]) { case (existingAssets, roadLink) =>
+      val assetsOnRoadLink = linearAssets.getOrElse(roadLink, Nil)
+      val adjustedAssets = fillOperations.foldLeft(assetsOnRoadLink) { case (currentSegments, operation) =>
+        operation(currentSegments)
+      }
+      existingAssets ++ adjustedAssets
+    }
+  }
+
+  def findStartEndRoadLinkOnChain(roadLinks: Seq[RoadLink]): Seq[(RoadLink, Option[Point], Option[Point])] = {
+    if (debbuger) println("findStartEndRoadLinkOnChain")
+    val borderRoadLinks = roadLinks.filterNot { r =>
+      val (first, last) = GeometryUtils.geometryEndpoints(r.geometry)
+      val RoadLinksFiltered = roadLinks.filterNot(_.linkId == r.linkId)
```

Comment:
Not a big problem, but usually variables start with a lower case letter. If all team agrees change it please.